### ISSUE TITLE
How to remove unnecessary non-sensitive scope from OAuth flow

### ISSUE THREAD

**Issue by ZYinMD** [google/google-api-javascript-client#731]
Simple example, using the Tasks API v1:
```js
// init:
gapi.auth2.init({
  clientId,
  scope: "https://www.googleapis.com/auth/tasks",
  fetch_basic_profile: false,
});

// start auth flow:
gapi.auth2.getAuthInstance().signIn();
```
I only need 1 scope, and here's what I expect to see (screenshot taken from other people's app):

![less scope](https://user-images.githubusercontent.com/32368482/114977118-2bd94480-9e55-11eb-9c41-335d690b34df.png)

However, in my own app, the OAuth window is broken into 2 steps, and always contains an additional scope in grey color, I don't need it and I don't how to remove it:

![more scope 1](https://user-images.githubusercontent.com/32368482/114976215-98ebda80-9e53-11eb-8da7-5ec173942f4d.png)
![more scope 2](https://user-images.githubusercontent.com/32368482/114976220-9ab59e00-9e53-11eb-8fda-b67aea086721.png)

From user's point of view, it's unclear what the grey scope means. I don't want my users to see it. How can I remove it?

I inspected the frontend js code of the app in img1, there's no significant difference from mine. The only difference might be the project setting on Google Cloud. However, I'm sure I only have 1 scope in my "OAuth Consent Screen" setting in Google Cloud, and my app is verified too.

**Comment by sarthak625** [comment]
The same is occurring for me too. There is no way i am able to remove openid from my scopes. Also, now the selected scopes are not ticked by default. Without this extra permission, the user atleast gets a prompt that no permissions are provided to the third party app. By having openid present, user will just click next and be clueless why the third party app isn't functioning.

**Comment by ZYinMD** [comment]
@sarthak625 I'm OP. After receiving an email notification about your reply, I tested it again today. It seems that my problem has been fixed – there's only one step in the process now, and there's no additional greyed-out checkbox that I didn't include. Not sure what you experienced, but I'm good for now.

**Comment by sarthak625** [comment]
@ZYinMD Are you using the same code? I am using gapi.client.init How are you calling google apis by using gapi.auth2.init .
For me it says gapi.client.calendar is undefined if i move to gapi.auth2.init. Also the consent window has permissions unticked by default now which is a headache as normally users dont tick the required permissions themselves.

**Comment by ZYinMD** [comment]
@sarthak625 Yes my code never changed. My original problem was that the oauth flow was broken into 2 consent windows, as shown in my screenshots. Now it seems only 1 consent window was showing, so my problem was solved. Are you still seeing 2 consent windows?

Yes indeed the checkbox was unchecked by default now, but it's not an issue in my case, because I request only 1 permission, and if the user doesn't check it, there's no way to click "next".

**Comment by sarthak625** [comment]
@ZYinMD For me the issue is that the second step has the greyed out openid permission and I can't find a way to remove it. I am also requesting only one permission from the user but gapi seems to be adding openid to the scope automatically.

Is your api. js file downloaded locally, or are you using the following CDN https://apis.google.com/js/api.js

I figure it is something gapi js is doing internally.

**Comment by ZYinMD** [comment]
@sarthak625 Mine was loaded from CDN. 

I did some other tests, and got the greyed openid scope again, (although it shows in the same window as the permission I'm requesting, no 2nd window.) 

Things seem inconsistent, I feel it may behave differently on localhost. Could you try [my production site](https://fullscreen-for-googletasks.com) and tell me if you see the greyed openid scope being requested?

**Comment by sarthak625** [comment]
@ZYinMD I can see it in your production site also. :(

Here is a screenshot
![image](https://user-images.githubusercontent.com/11756075/127274721-9b71f477-6d88-4482-8820-7dd8ea5dbc88.png)

**Comment by ZYinMD** [comment]
@sarthak625 

Thanks, so my bad, the issue was still the same. I have edited my previous replies.

In my tests 2 days ago, I was required to grant permission again, but didn't see the openid scope. Now I think about it, maybe it was because the test account I used already logged in many months ago. Maybe the permission to sensitive scopes need re-tick, but openid doesn't.

**Comment by xpt1x** [comment]
Were you guys able to find any workaround yet ? @ZYinMD @sarthak625 
I am experiencing same problem and my moto is similar to yours...

**Comment by sarthak625** [comment]
I was able to get rid of the extra permission by using gapi.client.authorize instead of gapi.auth2.init
https://developers.google.com/identity/sign-in/web/reference#gapiauth2authorizeparams_callback

However a drawback of this is that to logout you will need to refresh the screen.

**Comment by ZYinMD** [comment]
@sarthak625 are you able to stay logged-in for 1+ hours after this?

**Comment by sarthak625** [comment]
@ZYinMD I haven't tried that yet but i think it will be able to. However on a page reload the credentials will be lost and the user will need to authorize again. My use case is a very specific task which is expected to require google auth on every invocation so i can make use of gapi.client.authorize for my use case. Only problem i run into is that there is no way of calling gapi.client.logout or anything similar.

**Comment by ZYinMD** [comment]
@sarthak625 I asked whether you were able to stay logged in for 1+ hours, because the doc you linked says it fits "if your application only needs to requests a Google API endpoint once", which made me think it may not be able to handle token refresh.

Hope someone tries it out. I believe this method was new – this part of the doc wasn't there 4 months ago.

**Comment by sarthak625** [comment]
@ZYinMD As i said i didn't check remaining signed in for an hour as i needed to perform a one time action. However as suggested by the given documentation the access token is automatically refreshed, so you can try if it works for you.

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.
